autopkg/autopkg_tools.py

```diff
@@ -145,37 +185,51 @@ def checkout(branch, new=True):
 
 
 ### Recipe handling
-def handle_recipe(recipe):
-    recipe.run()
-
-    if recipe.results["imported"]:
-        checkout(recipe.branch)
-        for imported in recipe.results["imported"]:
-            git_run(["add", f"'pkgs/{ imported['pkg_repo_path'] }'"])
-            git_run(["add", f"'pkgsinfo/{ imported['pkginfo_path'] }'"])
-
-        git_run(
-            ["commit", "-m", f"'Updated { recipe.name } to { recipe.updated_version }'"]
-        )
-        git_run(["push", "--set-upstream", "origin", recipe.branch])
-
-    return recipe.results
+def handle_recipe(recipe, opts):
+    if not opts.disable_verification:
+        recipe.verify_trust_info()
+        if recipe.verified is False:
+            recipe.update_trust_info()
+    if recipe.verified in (True, None):
+        recipe.run()
+        if recipe.results["imported"]:
+            checkout(recipe.branch)
+            for imported in recipe.results["imported"]:
+                git_run(["add", f"'pkgs/{ imported['pkg_repo_path'] }'"])
+                git_run(["add", f"'pkgsinfo/{ imported['pkginfo_path'] }'"])
+            git_run(
+                [
+                    "commit",
+                    "-m",
+                    f"'Updated { recipe.name } to { recipe.updated_version }'",
+                ]
+            )
+            git_run(["push", "--set-upstream", "origin", recipe.branch])
+    return recipe
 
 
-def parse_recipes(file_path):
+def parse_recipes(recipes):
     recipe_list = []
-    ext = os.path.splitext(file_path)[1]
-
-    if ext == ".json":
-        parser = json.load
-    elif ext == ".plist":
-        parser = plistlib.load
+    ## Added this section so that we can run individual recipes
+    if RECIPE_TO_RUN:
```

Comment:
This is a bit 🍝 - parse_recipes shouldn't take paths based on if RECIPE_TO_RUN exists (ie inferring the value of its parameter by inspecting the existence of some top-level parameter invites confusion)

Instead, let's use typing:

```
def parse_recipes(recipes):
    recipe_list = []
    if isinstance(recipes, list):
        # parse each recipe expecting it to reference a single recipe override as in RECIPE_TO_RUN
    elif isinstance(recipes, str) and os.path.exists(recipes):
        # parse as if its a file containing a list, with parser selection
    else:
        raise ValueError(f"Recipes must be a list of recipes or a reference to a file containing such a list. Got {recipes}: {repr(type(recipes))}")
```

Reply:
Changes have been implemented.